instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpClientTracer.java

```diff
@@ -60,60 +59,67 @@ protected HttpClientTracer(Tracer tracer) {
     super(tracer);
   }
 
-  public Span startSpan(REQUEST request) {
-    return startSpan(request, -1);
+  public boolean shouldStartSpan(Context parentContext) {
+    return parentContext.get(CONTEXT_CLIENT_SPAN_KEY) == null;
   }
 
-  public Span startSpan(REQUEST request, long startTimeNanos) {
-    return internalStartSpan(request, spanNameForRequest(request), startTimeNanos);
+  public Context startSpan(Context parentContext, REQUEST request, CARRIER carrier) {
+    return startSpan(parentContext, request, carrier, -1);
   }
 
-  public Scope startScope(Span span, CARRIER carrier) {
-    Context context = Context.current().with(span);
-
+  public Context startSpan(
+      Context parentContext, REQUEST request, CARRIER carrier, long startTimeNanos) {
+    Span span =
+        internalStartSpan(parentContext, request, spanNameForRequest(request), startTimeNanos);
     Setter<CARRIER> setter = getSetter();
     if (setter == null) {
       throw new IllegalStateException(
           "getSetter() not defined but calling startScope(), either getSetter must be implemented or the scope should be setup manually");
     }
+    Context context = parentContext.with(span).with(CONTEXT_CLIENT_SPAN_KEY, span);
     OpenTelemetry.getGlobalPropagators().getTextMapPropagator().inject(context, carrier, setter);
-    context = context.with(CONTEXT_CLIENT_SPAN_KEY, span);
-    return context.makeCurrent();
+    return context;
```

Comment:
Is it a good idea to inject new context into outgoing request during `startSpan`? Technically, creating a new span does not make it "active" yet. It may never become one and then propagating its context is wrong. In this sense old `startScope` method was much better suited for context propagation.

Reply:
I don't think injecting context into outgoing http requests is related to putting the context in scope, e.g. async http clients never put the context in scope